tests/extensions/test_app.py

```diff
@@ -404,6 +404,20 @@ def test_flask_merge_charm_libs(flask_input_yaml, tmp_path):
         {"config": {"options": {"flask-foobar": {"type": "string"}}}},
         id="reserved-config-prefix-flask",
     ),
+    pytest.param(
+        {
+            "config": {
+                "options": {
+                    "non-optional": {
+                        "type": "string",
+                        "optional": False,
+                        "default": "default value",
+                    }
+                }
+            }
+        },
+        id="non-optional-config-with-defautl",
```

Comment:
Let's fix the typo

Should we add a test case with a optional: False without a default?